src/audio/audio.cpp

```diff
@@ -122,67 +132,68 @@ void PipeAudio::SetPosition(std::uint64_t position)
 	this->ClearFrame();
 }
 
-void PipeAudio::ClearFrame()
+void Basic_audio::ClearFrame()
 {
 	this->frame.clear();
-	this->frame_iterator = this->frame.end();
+	this->frame_span = gsl::span<std::byte, 0>();
 }
 
-Audio::State PipeAudio::Update()
+Audio::State Basic_audio::Update()
 {
-	assert(this->sink != nullptr);
-	assert(this->src != nullptr);
+	Expects(this->sink != nullptr);
+	Expects(this->src != nullptr);
 
-	bool more_available = this->DecodeIfFrameEmpty();
+	const auto more_available = this->DecodeIfFrameEmpty();
 	if (!more_available) this->sink->SourceOut();
 
 	if (!this->FrameFinished()) this->TransferFrame();
 
-	return this->sink->State();
+	return this->sink->CurrentState();
 }
 
-void PipeAudio::TransferFrame()
+void Basic_audio::TransferFrame()
 {
-	assert(!this->frame.empty());
-	assert(this->sink != nullptr);
-	assert(this->src != nullptr);
+	Expects(!this->frame.empty());
+	Expects(this->sink != nullptr);
+	Expects(this->src != nullptr);
 
-	this->sink->Transfer(this->frame_iterator, this->frame.end());
+	auto written = this->sink->Transfer(this->frame_span);
+	this->frame_span =
+	        this->frame_span.last(this->frame_span.size() - written);
```

Comment:
this isn't 1980, line lengths longer than 80 is fine (i like a soft limit of 120, personally, though 150 works too)

Reply:
Agree.  There are some bits in the tests where `clang-format` has utter kittens trying to deal with a combination of macro soup and line lengths, which this might help with